platforms/gke/base/use-cases/inference-ref-arch/kubernetes-manifests/online-inference-gpu-h100-llama4-scout/online-inference-gpu-deployment.yaml

```diff
@@ -0,0 +1,213 @@
+            cpu: "64"
+            memory: "512G"
+            nvidia.com/gpu: "8"
+    restartPolicy: Never
+---
+apiVersion: autoscaling.x-k8s.io/v1beta1
+kind: ProvisioningRequest
+metadata:
+  name: vllm-h100
+spec:
+  provisioningClassName: queued-provisioning.gke.io
+  parameters:
+    maxRunDurationSeconds: "86400"
+  podSets:
+    - count: 1
+      podTemplateRef:
+        name: vllm-h100
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: vllm-h100
+spec:
+  selector:
+    matchLabels:
+      app: vllm-h100
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: vllm-h100
+      annotations:
+        gke-gcsfuse/volumes: "true"
+        gke-gcsfuse/cpu-limit: "0"
+        gke-gcsfuse/memory-limit: "0"
+        gke-gcsfuse/ephemeral-storage-limit: "0"
+        cluster-autoscaler.kubernetes.io/consume-provisioning-request: vllm-h100
+        cluster-autoscaler.kubernetes.io/provisioning-class-name: "queued-provisioning.gke.io"
+    spec:
+      containers:
+        # Load safetensors files in the gcsfuse cache
+        - name: fetch-safetensors
+          image: busybox
+          command: ["/bin/sh", "-c"]
+          args:
+            - |
+              echo "########### $(date) - Starting parallel-fetch-safetensors for model: ${MODEL_ID}"
+              ls -alR /gcs
+              find /gcs/${MODEL_ID}/*safetensors -type f | xargs -I {} -P 15 sh -c 'echo "########### $(date) - Fetching: {}"; dd if={} of=/dev/null'
+              echo "########### $(date) - Finished parallel-fetch-safetensors"
+              sleep infinity
+          env:
+            - name: MODEL_ID
+              valueFrom:
+                configMapKeyRef:
+                  key: MODEL_ID
+                  name: model-download-configmap
+          volumeMounts:
+            - name: gcsfuse
+              mountPath: /gcs
+              readOnly: true
+        - name: inference-server
+          args:
+            - --disable-log-requests
+            - --gpu-memory-utilization=0.95
+            - "--model=/gcs/$(MODEL_ID)"
+            - --tensor-parallel-size=8
+            - --trust-remote-code
+            - --max-model-len=1000000
+          env:
+            - name: MODEL_ID
+              valueFrom:
+                configMapKeyRef:
+                  key: MODEL_ID
+                  name: model-download-configmap
+          image: vllm/vllm-openai:v0.8.4
+          imagePullPolicy: Always
+          readinessProbe:
+            failureThreshold: 6000
+            httpGet:
+              path: /health
+              port: 8000
+              scheme: HTTP
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          ports:
+            - containerPort: 8000
+              name: metrics
+          resources:
+            requests:
+              cpu: "64"
+              memory: "512G"
+              nvidia.com/gpu: "8"
```

Comment:
When you change the model to use the required hardware, you will also need to change resource requests

Reply:
I've seen you have a specific folder per model. That is fine. But we may be able to have a single yaml if we make these things configurable

Reply:
Same as above, let's keep this open as a reminder.

Reply:
Moved to a patch